components/src/dynamo/vllm/tests/test_vllm_worker_handler.py

```
@@ -388,6 +388,114 @@ async def fake_generate(*args, **kwargs):
+        """`ExampleHiddenStatesConnector` (and any other KV connector) attaches
+        its per-request payload to `RequestOutput.kv_transfer_params`; the base
+        `generate_tokens` must forward it under `engine_data["kv_transfer_params"]`
+        on the final chunk so the Rust response builder can surface it under
+        `nvext.kv_transfer_params`. Regression for the extract_hidden_states path
+        the speculators capture flow depends on."""
+        from vllm.sampling_params import SamplingParams
+
+        handler = _make_handler()
+        handler._extract_logprobs = MagicMock(return_value=(None, None))
+
+        kv_payload = {"hidden_states_path": "/tmp/hs/req-kv-1.safetensors"}
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/hs/req-kv-1.safetensors"

(S108)